crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -205,13 +208,34 @@ where
                                 ));
                             };
 
+                            let nsec3s = verified_message
+                                .name_servers()
+                                .iter()
+                                .filter(|rr| is_dnssec(rr, RecordType::NSEC3))
+                                .collect::<Vec<_>>();
```

Comment:
Could we instead do the `as_dnssec().as_nsec3()` thing here so that we have a stronger/clearer invariant across the `verify_nsec3()` function call boundary?